### PR TITLE
Add Modern C++ DL example

### PR DESCRIPTION
**Pull request by luisrg-rti**
This example requires changes to the `cmake-utils` sub-repository before it can be compiled

### CODE REVIEW

**Comment by lulivi** [comment]
Hey @luisrg-rti can you move the source code to the main directory similarly to all the other examples? https://github.com/rticommunity/rticonnextdds-examples/tree/develop/examples/connext_dds/asyncwaitset/c%2B%2B11 same with the header file.

**Comment by lulivi** [comment]
Also take a look to the linting https://github.com/rticommunity/rticonnextdds-examples/actions/runs/6633405661/job/18021065371?pr=632 and formatting https://github.com/rticommunity/rticonnextdds-examples/actions/runs/6633405665/job/18021065354?pr=632.

File: examples/distributed_logger/c++11/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+find_package(
+    RTIConnextDDS "7.2.0"
+    REQUIRED
+    COMPONENTS
+      core
+      distributed_logger
+)
+
+# Include ConnextDdsAddExample.cmake from resources/cmake
+include(ConnextDdsAddExample)
```
Comment (lulivi) [inline examples/distributed_logger/c++11/CMakeLists.txt:31]:
```suggestion
```

This is not used here

File: examples/distributed_logger/c++11/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+#  RTI is under no obligation to maintain or support the Software.  RTI shall
+#  not be liable for any incidental or consequential damages arising out of the
+#  use or inability to use the software.
+#
+cmake_minimum_required(VERSION 3.11)
+project (ModernCppDistLogger)
+
+# Find RTI Connext dependencies
+list(APPEND CMAKE_MODULE_PATH
+    "${CMAKE_CURRENT_SOURCE_DIR}/../../../resources/cmake/Modules"
```
Comment (lulivi) [inline examples/distributed_logger/c++11/CMakeLists.txt:17]: Remember to add an extra `../` when this is moved to `examples/connext_dds`

**Comment by alexcamposruiz** [comment]
Make the example consistent with the rest:

1. Move to connext_dds
2. Create a top-level README with language-independent description of the example, and mention that for other languages the examples are shipped (see the basic_security example for ex)
3. Move all sources to c++11 (remove src and include dirs)
4. Make the c++11/README.md focus on how to run the c++11 example only.

**Comment by luisrg-rti** [comment]
> Make the example consistent with the rest:
> 
>     1. Move to connext_dds
> 
>     2. Create a top-level README with language-independent description of the example, and mention that for other languages the examples are shipped (see the basic_security example for ex)
> 
>     3. Move all sources to c++11 (remove src and include dirs)
> 
>     4. Make the c++11/README.md focus on how to run the c++11 example only.

I'll make these changes. I think I may have followed the structure of the other DL examples in `connextdds/examples.1.0` by mistake.

**Comment by lulivi** [comment]
Hi @luisrg-rti can you add your `distributed_logger` directory to here? https://github.com/rticommunity/rticonnextdds-examples/blob/master/examples/connext_dds/CMakeLists.txt#L108

**Comment by lulivi** [comment]
Once the [cmake-utils PR](https://github.com/rticommunity/rticonnextdds-cmake-utils/pull/109) is merged, we will launch a final build (after the `examples/connext_dds/CMakeLists.txt` change is done) to check everything working as expected

File: examples/connext_dds/distributed_logger/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+## Example Description
+
+Similar examples are available in other programming languages in the
+`rti_workspace/<version>/examples/distributed_logger/<language>/`
+folder created under your home directory when RTI Connext is installed.
+
+This example shows a simple application running Distributed Logger to
+log several messages using different configurations.
+
+The log messages can then be visualized using RTI Tools such as RTI Spy
```
Comment (alexcamposruiz) [inline examples/connext_dds/distributed_logger/README.md:20]: Can you add a shell ```-block showing how to run spy and the expected output?

**Review by alexcamposruiz — CHANGES_REQUESTED** [review]

**Review by alexcamposruiz — APPROVED** [review]

**Comment by lulivi** [comment]
Seems to be working fine:
```
2023-11-17T07:29:55.8992829Z [ 25%] Building CXX object connext_dds/distributed_logger/c++11/CMakeFiles/DistLoggerExample.dir/DistLoggerExample.cxx.o 
2023-11-17T07:29:57.0808939Z [ 25%] Linking CXX executable DistLoggerExample                        
2023-11-17T07:29:57.1604108Z [ 25%] Built target DistLoggerExample
```

**Pull request merged**